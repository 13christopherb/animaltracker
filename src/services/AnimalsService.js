const api = 'http://localhost:5000';
//const api = 'https://mbo-animal-tracker-api.herokuapp.com'

const bearer = 'Bearer '+ localStorage.getItem('accessToken');
const headers = new Headers({
    'Accept': 'application/json',
    'Authorization': bearer,
    'Content-Type': 'application/json',
    'mode':  "cors"
});



function handleErrors(response) {
    if (!response.ok) {
        throw Error(response.statusText);
    }
    return response;
}

export const getAllAnimals = () =>
    fetch(api+ '/animals', {
        method: 'GET',
        withCredentials: true,
        headers: headers
    }).then(handleErrors).then(res => res.json());

export const addAnimal = (animal) =>
    fetch(api + `/animals`, {
        method: 'POST',
        headers: headers,
        body: JSON.stringify(animal)
    }).then(handleErrors).then(res => res.json());

export const deleteAnimal = (id) =>
    fetch(api + '/animal/' + id, {
        method: 'DELETE',
        headers: headers,
        body: id
    }).then(handleErrors).then(res => res.json());