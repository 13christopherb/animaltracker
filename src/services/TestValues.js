export const animal1 = {
    name: 'Edginald',
    species: 'CSL',
    weight: 14,
    isGettingTubed: true,
    isGettingControlledMeds: false,
    id: 1,
    location: 'NRO'
}

export const animal2 = {
    name: 'Kirt',
    species: 'CSL',
    weight: 85,
    isGettingTubed: true,
    isGettingControlledMeds: false,
    id: 2,
    location:'MBO'
}

export const animal3 = {
    name: 'Pipester',
    species: 'NFL',
    weight: 10,
    isGettingTubed: true,
    isGettingControlledMeds: false,
    id: 3,
    location: 'SLO'
}

export const animal4 = {
    name: 'Spud',
    species: 'ES',
    weight: 34,
    isGettingTubed: true,
    isGettingControlledMeds: false,
    id: 4,
    location: 'MBO'
}

export const animalsMinusOne = [animal2, animal3, animal4]
export const animals = [animal1, animal2, animal3, animal4]
