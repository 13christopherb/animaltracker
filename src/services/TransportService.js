const api = 'http://localhost:5000'
//const api = 'https://mbo-animal-tracker-api.herokuapp.com'

const headers = {
    'Accept': 'application/json',
    'Content-Type': 'application/json',
    'mode':  "cors"
};

function handleErrors(response) {
    if (!response.ok) {
        throw Error(response.statusText);
    }
    return response;
}

export const getAllTransports = () =>
    fetch(api+ '/transports', {
        method: 'GET',
        headers: headers
    }).then(handleErrors).then(res => res.json());

export const addTransport = (transport) =>
    fetch(api + `/transports`, {
        method: 'POST',
        headers: headers,
        body: JSON.stringify(transport)
    }).then(handleErrors).then(res => res.json());

export const deleteAnimal = (id) =>
    fetch(api + '/transports/' + id, {
        method: 'DELETE',
        headers: headers,
        body: id
    }).then(handleErrors).then(res => res.json());