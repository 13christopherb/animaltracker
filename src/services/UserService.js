const api = 'http://localhost:5000';
const headers = {
    'Accept': 'application/json',
    'Content-Type': 'application/json',
    'mode': "cors"
};

export const login = (login) =>
    fetch(api + '/login', {
        method: 'POST',
        headers: headers,
        body: JSON.stringify(login)
    }).then(handleResponse)
        .then(user => {
            if (user.accessToken) {
                localStorage.setItem('user', JSON.stringify(user));
            }

            return user;
        });

export const logout = () =>
    localStorage.removeItem('user');

export const register = (login) =>
    fetch(api + '/registration', {
        method: 'POST',
        headers: headers,
        body: JSON.stringify(login)
    }).then(res => res.json());

function handleResponse(response) {
    return response.text().then(text => {
        const data = text && JSON.parse(text);
        if (!response.ok) {
            if (response.status === 401) {
                // auto logout if 401 response returned from api
                logout();
            }
            const error = (data && data.message) || response.statusTest;
            return Promise.reject(error);
        }

        return data;
    })
}