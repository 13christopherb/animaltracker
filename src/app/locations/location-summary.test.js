import React from 'react';
import {shallow, mount} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import LocationSummary from './location-summary';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk'
import * as testValues from '../../services/TestValues'

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

let location = {
    locationName: 'MBO',
    animals: [{
        name: 'Kirt',
        species: 'CSL',
        weight: 85,
        isGettingTubed: true,
        isGettingControlledMeds: false,
        id: 2,
        location: 'MBO'
    }
    ]
};

const initialState = {
    animals: testValues.animals,
    locations: {locations: {
            'MBO': location
        }}
};


describe('<LocationSummary />', () => {
    describe('render()', () => {
        test('should render the component', () => {
            const wrapper = shallow(<LocationSummary expanded={false}
                                                     expandSummary={jest.fn()}
                                                     animals={location.animals}
                                                     locationName={location.locationName}

            />);

            expect(toJson(wrapper)).toMatchSnapshot()
        })
    })
});