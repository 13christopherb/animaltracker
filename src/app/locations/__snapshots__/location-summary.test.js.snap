// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationSummary /> render() should render the component 1`] = `
<div
  className="card"
>
  <div
    className="card-body h-100"
  >
    <h5
      className="card-title"
    >
      MBO
      <button
        className="btn btn-sm btn-outline-success float-right"
        data-toggle="dropdown"
        id="dropdownTitleButton"
        type="button"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="plus"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
      <div
        aria-labelledby="dropdownTitleButton"
        className="dropdown-menu"
      >
        <button
          className="btn btn-sm dropdown-item"
          data-target="#MBOmodalForm"
          data-toggle="modal"
          type="button"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="otter"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
           Add animal
        </button>
        <button
          className="btn btn-sm dropdown-item"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="truck-pickup"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
           Add Transport
        </button>
      </div>
    </h5>
    <ul
      className="list-group list-group-flush"
    >
      <ReduxForm
        animal={
          Object {
            "id": 2,
            "isGettingControlledMeds": false,
            "isGettingTubed": true,
            "location": "MBO",
            "name": "Kirt",
            "species": "CSL",
            "weight": 85,
          }
        }
        key="2"
      />
    </ul>
  </div>
  <footer
    className="card-footer bg-transparent"
  />
  <ReduxForm
    location="MBO"
    toggleAddAnimal={[Function]}
  >
    <AnimalModalForm
      id="MBOmodalForm"
    />
  </ReduxForm>
</div>
`;
